### PR TITLE
Python instrumentation featuregate to cli

### PR DESCRIPTION
**Pull request by changexd**
**Description:** <Describe what has changed.>

change python instrumentation into command line

**Link to tracking Issue(s):** <Issue number if applicable>

- Resolves: #2672

### CODE REVIEW

File: .chloggen/featuregate_cli_python.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: breaking
+
+# The name of the component, or a single word describing the area of concern, (e.g. collector, target allocator, auto-instrumentation, opamp, github action)
+component: operator
+
+# A brief description of the change. Surround your text with quotes ("") if it needs to start with a backtick (`).
+note:  remove featuregate `EnablePythonAutoInstrumentationSupport`. Use command line flag `--enable-python-instrumentation` instead
```
Comment (pavolloffay) [inline .chloggen/featuregate_cli_python.yaml:8]: What was the `flag` what we are removing?

Comment (TylerHelmuth) [reply .chloggen/featuregate_cli_python.yaml:8]: Should be `operator.autoinstrumentation.python`

File: main.go [hunk 2]
```diff
@@ -136,6 +137,7 @@ func main() {
 	pflag.BoolVar(&enableMultiInstrumentation, "enable-multi-instrumentation", false, "Controls whether the operator supports multi instrumentation")
 	pflag.BoolVar(&enableApacheHttpdInstrumentation, constants.FlagApacheHttpd, true, "Controls whether the operator supports Apache HTTPD auto-instrumentation")
 	pflag.BoolVar(&enableDotNetInstrumentation, constants.FlagDotNet, true, "Controls whether the operator supports dotnet auto-instrumentation")
+	pflag.BoolVar(&enablePythonInstrumentation, constants.FlagPython, true, "Controls whether the operator supports python auto-instrumentation")
```
Comment (yuriolisa) [inline main.go:140]: Should this instrumentation be disabled by default?

Comment (swiatekm-sumo) [reply main.go:140]: It was a beta feature flag, and therefore enabled by default, so it should stay enabled imo.

Comment (pavolloffay) [reply main.go:140]: enabled by default sounds good

**Review by pavolloffay — APPROVED** [review]

**Review by swiatekm-sumo — APPROVED** [review]

**Pull request merged**